src/i18n/Data/Intl/IntlLocales.php

```diff
@@ -0,0 +1,1622 @@
+            return static::$cache_languages[$locale];
+        }
+
+        // Localise all languages
+        $languages = $this->config()->get('languages');
+        $localised = [];
+        foreach ($languages as $code => $default) {
+            $localised[$code] = $this->languageName($code);
+        }
+
+        // Save cache
+        static::$cache_languages[$locale] = $localised;
+        return $localised;
+    }
+
+    /**
+     * Get name of locale
+     *
+     * @param string $locale
+     * @return string
+     */
+    public function localeName($locale)
+    {
+        return Locale::getDisplayName($locale, i18n::get_locale());
+    }
+
+    /**
+     * Get language name for this language or locale code
+     *
+     * @param string $code
+     * @return string
+     */
+    public function languageName($code)
+    {
+        return Locale::getDisplayLanguage($code, i18n::get_locale());
+    }
+
+    /**
+     * Cache of localised countries, keyed by locale localised in
+     *
+     * @var array
+     */
+    private static $cache_countries = [];
+
+    /**
+     * Get all country codes and names
+     *
+     * @return array Map of country code => name
+     */
+    public function getCountries()
+    {
+        // Cache by locale
+        $locale = i18n::get_locale();
+        if (!empty(static::$cache_countries[$locale])) {
+            return static::$cache_countries[$locale];
+        }
+
+        // Localise all countries
+        $countries = $this->config()->get('countries');
+        $localised = [];
+        foreach ($countries as $code => $default) {
+            $localised[$code] = $this->countryName($code);
+        }
+
+        // Always sort by localised name, not key
+        $collator = new Collator(i18n::get_locale());
+        $collator->asort($localised);
+        static::$cache_countries[$locale] = $localised;
+        return $localised;
+    }
+
+    /**
+     * Get name of country
+     *
+     * @param string $code ISO 3166-1 country code
+     * @return string
+     */
+    public function countryName($code)
+    {
+        return Locale::getDisplayRegion('-' . $code, i18n::get_locale());
+    }
+
+    /**
+     * Returns the country code / suffix on any locale
+     *
+     * @param string $locale E.g. "en_US"
+     * @return string Country code, e.g. "us"
+     */
+    public function countryFromLocale($locale)
+    {
+        return strtolower(Locale::getRegion($locale)) ?: null;
+    }
+
+    /**
+     * Validates a "long" locale format (e.g. "en_US") by checking it against {@link $locales}.
+     *
+     * @param string $locale
+     * @return bool
+     */
+    public function validate($locale)
+    {
+        if (!$locale) {
+            return false;
+        }
+        $lang = $this->langFromLocale($locale);
+        $region = $this->countryFromLocale($locale);
+        if (!$lang || !$region) {
+            return false;
+        }
+        return strcasecmp($lang, $region)
```

Comment:
Is multiplication really what you want here? Seems like a roundabout way of expressing a conditional